File: training/python/de/sots/cellarsandcentaurs/domain/model/armor_class.py
from de.sots.cellarsandcentaurs.application.creature_util import CreatureUtil


class ArmorClass:
    def __init__(self, base: int, bonus: int, description: str = CreatureUtil.STANDARD_ARMOR_CLASS_DESCRIPTION):
        self.description = description
        self.base = base
        self.bonus = bonus

    @property
    def total(self) -> int:
        return self.base + self.bonus

File: training/kotlin/src/main/kotlin/de/sots/cellarsandcentaurs/domain/model/ArmorClass.kt
package de.sots.cellarsandcentaurs.domain.model

import de.sots.cellarsandcentaurs.application.CreatureUtil

data class ArmorClass(
    val base: Int,
    val bonus: Int,
    val description: String = CreatureUtil.STANDARD_ARMOR_CLASS_DESCRIPTION
) {
    val total: Int
        get() = base + bonus
}
